### ISSUE TITLE
Critical issue: EML import functionality completely broken

### ISSUE THREAD

**Issue by de04080** [thundernest/import-export-tools-ng#613]
### Description

Importing messages in EML format is broken. I have exported messages from Outlook.com, iCloud.com, and other services. I initially reported this issue in bug report #546, but after reviewing the comments, I realized that the problem is related to endless header content being added to the messages, during the import of EML messages back into Thunderbird. Therefore, I am opening a separate bug report for this issue. Thank you.

### System Information

- Thunderbird 128
- Import Export Tools NG 14.12
- Windows 11 Pro

### Exporting to

- MBOX = fine
- HTML = fine
- Plain Text = fine
- EML = **broken!**

### Details

A folder containing 10 emails and 2 attachments is 286 KB. After exporting it to EML and re-importing it, the size increases to 3.9 MB. Most of the messages are cluttered with extensive header content, as shown in the attached screenshot below, and they no longer resemble the originals.

![SCR_20240803_175035](https://github.com/user-attachments/assets/3c059b13-62e5-47b4-aead-b306ac2d8e14)

None of the options in the add-on made a difference. Exporting each email individually versus exporting the full folder did not change the outcome. Threading and unthreading the emails before exporting also had no effect.

**Comment by cleidigh** [comment]
@de04080 
I have a beta for this issue. So far my testing looks good doing imports up to 20k eml messages. Let me know how it looks on your side.

https://github.com/thunderbird/import-export-tools-ng/blob/v14.1.3/xpi/beta/import-export-tools-ng-14.1.3-b2-tb.xpi

@cleidigh

**Comment by cleidigh** [comment]
@de04080 
This is a github quirk. See below for tip
https://github.com/thunderbird/import-export-tools-ng/issues/595
@cleidigh

**Comment by kmbuthia** [comment]
Hello @cleidigh 

I also installed this "patched" version of the addon and can confirm that bulk import eml files no longer works. It did some weeks ago but not any more unfortunately.
At the moment, uploading one eml file at a time is the only thing that seems to work. If there's another bulk import option/approach that can be used as a work-around, please advise and thanks a lot in advance for the guidance/support.

<img width="923" alt="Screenshot 2024-08-06 at 13 51 11" src="https://github.com/user-attachments/assets/2568cdeb-49c0-4b97-86d7-32194580d84e">

**Comment by cleidigh** [comment]
@de04080 
You are correct, you should not see a more than double difference. The mbox after import will be slightly larger with some additional headers, but minimal.
So far I cannot reproduce what you see. I did multiple export imports with small and multi megabyte folders all with expected results.
Can you try just a few messages and see if you get such different sizes? Then maybe we can analyze.
@cleidigh

**Comment by cleidigh** [comment]
@kmbuthia
Can you id what Thunderbird version?
v14.1.3-b2 should fix for Thunderbird v128, however, it won't work on v115. I am doing a rev that will work on both now.
Can you be more specific about what is happening?
What was working before?
@cleidigh

**Comment by kmbuthia** [comment]
@cleidigh Thunderbird version: 128.1.0esr (64 bit - on a Mac)

Steps to reproduce eml import failure:
1. I exported a mailbox with this plugin and chose eml as the export option (roughly 100 email threads. Total size less than 1GB)
2. I then attempted to bulk import the eml messages into a new inbox using the eml import option

Issue/error:
I was unable to import more than one eml message at a time. Any attempt to use the "import all eml messages from a directory" or "import eml message from a directory with subdirectories" ended up importing one message then hanging indefinitely after that (waited 30min before killing the operation).
As mentioned before, what did work was choosing the "import individual eml messages" option then choosing a single eml file. Even with this option choosing just two emls would result in one eml being successfully uploaded and then the process hangs and the second eml doesn't get imported.

Hope this helps and let me know if I can provide more info.

**Comment by cleidigh** [comment]
@kmbuthia 
So I'm still trying to understand if importing multiple eml messages ever worked with v128esr? I see that you did install v14.1.3-b2 and that was not working.
I may have to add debug as this should work.
@cleidigh

**Comment by cleidigh** [comment]
@de04080 
This still perplexes me. I really think I need an example of your inputs and import results. Can you try to identify a couple of non personal eml messages that when imported are larger. Then if you can send me those and the resulting mbox to my test account 
test1@kokkini.net

To get the mbox you can choose Open folder directory and locate the mbox for the folder where you imported the couple emls. The mbox will have no extension. This way I can really compare things.
@cleidigh

**Comment by aryanrajseo** [comment]
Hi, i am trying to import eml files and it import 1 and then got stuck. Using Win 11 Pro 22H2, latest version.


<img width="1269" alt="image" src="https://github.com/user-attachments/assets/6305fff9-db2a-4aa5-9bc8-0114b74ecb69">

**Comment by sinnfeinn** [comment]
> Hi, i am trying to import eml files and it import 1 and then got stuck. Using Win 11 Pro 22H2, latest version.
> 
> 
> 
> 
> 
> <img width="1269" alt="image" src="https://github.com/user-attachments/assets/6305fff9-db2a-4aa5-9bc8-0114b74ecb69">
> 
> 

Same issue

**Comment by kmbuthia** [comment]
The screenshots above is also what I experienced @cleidigh while trying to use the import.

I just noticed that @de04080 screenshot shows uploading to local folders. However in my case (and I suspect it's the same case for sinnfein and aryanrajseo) we are trying to upload/import to an email server, not a local folder...and that doesn't seem to be working.

Let me know if this helps or if the plugin only supports import to local folders.

**Comment by cleidigh** [comment]
@kmbuthia 
Did you try b2, that addresses both local folders and imap.
https://github.com/thunderbird/import-export-tools-ng/blob/v14.1.3/xpi/beta/import-export-tools-ng-14.1.3-b2-tb.xpi
@cleidigh

**Comment by cleidigh** [comment]
@kmbuthia 
Working on it...
@cleidigh

**Comment by kmbuthia** [comment]
> It might be that you were trying to import to an imap folder? Try a local folder With b2.

Thanks @cleidigh and yes to answer the above question you asked some time ago...I was indeed trying to import into an imap folder/email server. Local folder import certainly works...but upload to IMAP server does not work unfortunately.

**Comment by cleidigh** [comment]
@kmbuthia 
@sinnfeinn 
@de04080 
Ok I just posted b3 with a specific fix for imap eml imports. I don't know why I missed this, but I did.
Besides various individual imports, I did a folder import of over 1500 messages. Since it's imap it's slower depending on your server etc.

https://github.com/thunderbird/import-export-tools-ng/blob/v14.1.3/xpi/beta/import-export-tools-ng-14.1.3-b3-tb.xpi

@cleidigh

**Comment by kmbuthia** [comment]
> https://github.com/thunderbird/import-export-tools-ng/blob/v14.1.3/xpi/beta/import-export-tools-ng-14.1.3-b3-tb.xpi

@cleidigh some good news on my end.

The last patch you shared above does indeed work. I am able to import multiple emails once more into an email server (IMAP server). Ran a test with ~250 emails and they all got uploaded smoothly.

Thanks a lot for the fix! 👍🏽
